### PR TITLE
Update issue templates to use the new tasklist markdown

### LINKED ISSUE #2538: Update issue templates to use the new tasklist markdown
## Description and context
New markdown for extended task lists is available: https://docs.github.com/en/issues/tracking-your-work-with-issues/about-tasklists

## Definition of done
Tracking issue template is updated

### PR DESCRIPTION
**Pull request by aadam-10**
closes #2538

Updates the tracking issue template to use the new tasklist markdown that synergizes better with Projects

### CODE REVIEW

**Review by hendrikhofstadt — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/tracking_issue.md [hunk 1]
```diff
@@ -11,19 +11,15 @@ This issue is for grouping *ADD_COMPONENT* related tasks that are necessary for
...
 <!--- Link related tracking issues within the project for easier navigation. -->
+<!--- Assign tasks to the appropriate tracking issue if there is more than one. -->
 
 - #XXXX
 - #XXXX
 - #XXXX
 
+<!--- Subtasks MUST be inside the ``` -> ``` code block. -->
+```[tasklist]
 ### Task list
```
Comment (hendrikhofstadt) [inline .github/ISSUE_TEMPLATE/tracking_issue.md:22]: Should the header be outside the code block?

**Review by hendrikhofstadt — CHANGES_REQUESTED** [review]
See inline comments

**Review by hendrikhofstadt — APPROVED** [review]

**Comment by bruce-riley** [comment]
Looks like a rebase went awry on this?

**Comment by aadam-10** [comment]
seems like, thanks for the headsup - I'll make a fresh pr bc I'm not sure how to fix it

**Pull request closed without merging**